src/gui/menus/PlayerMenu.cpp

```diff
@@ -68,24 +75,40 @@ static int changeFile(const std::string &filepath, playbackInfo_t* playbackInfo)
 	if(!playbackInfo)
 		return 0;
 
-	playbackInfo->usePlaylist = 0;
-
 	std::string extension = filepath.substr(filepath.find_last_of('.') + 1);
 	if (!strncmp(extension.c_str(), "json", 4)) {
 		std::string url;
 		Cfg::ParseNC(filepath.c_str(), &url);
-		playbackInfo->filepath = url;
+		// playbackInfo->filepath = url;
 	} else if (!strncmp(extension.c_str(), "pls", 3)) {
 		// Note to future me.
 		// Add pls support to netfmt.
-		Pls::Parse(filepath, &playbackInfo->playlistfile);
-		playbackInfo->usePlaylist = 1;
+		Pls::Parse(filepath, &playbackInfo->playlist);
 	} else if (!strncmp(extension.c_str(), "m3u", 3)) {
-		M3u::Parse(filepath, &playbackInfo->playlistfile);
-		playbackInfo->usePlaylist = 1;
-	} else
-		playbackInfo->filepath = filepath;
+		M3u::Parse(filepath, &playbackInfo->playlist);
+	} else {
+		playbackInfo->playlist.files.clear();
+		playbackInfo->playlist.files.shrink_to_fit();
+		
+		int idx = 0;
+		for (int i=0; i < expInst->Size(); i++) { 
+			// rule out unsupporteds
+			auto e = expInst->GetEntry(i);
+			std::string ext = e.filename.substr(e.filename.find_last_of(".") + 1);
+			if (e.directory || e.filename == ".." || !is_extension_supported(ext)) {
+				DEBUG(DSTRFY2(e.filename) "wasn't recognised as supported, continuing\n");
```

Comment:
Not sure what you think you are accomplishing by using `DSTRFY2()`, but it probably isn't what is actually happening.
What you want it probably `DEBUG("%s wasn't recognised as supported, continuing\n", e.filename.c_str());` (debug supports printf syntax).